src/main/java/com/microsoft/azure/vmagent/AzureVMCloud.java

```diff
@@ -483,6 +493,33 @@ public AzureVMAgentTemplate getAzureAgentTemplate(Label label) {
         }
         return null;
     }
+    
+    private static void putVariable(JsonNode template, String name, String value) {
+        ObjectNode.class.cast(template.get("variables")).put(name, value);
+    }
+    
+    private String enableUAMI(String vmBaseName, String uamiID, String location) throws IOException{
+        
+        InputStream embeddedTemplate = null;
+        JsonNode tmp = null;
+        try {
```

Comment:
should be able to use try with resources here so you don't need the finally block

```suggestion
        try (InputStream  embeddedTemplate = AzureVMManagementServiceDelegate.class.getResourceAsStream(EMBEDDED_TEMPLATE_UAMI);) {
```